bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,656 @@
+    private ScheduledFuture<?> logRefreshJob;
+
+    private RadioThermostatData rthermData = new RadioThermostatData();
+    
+    private Gson gson;
+
+    private int initialized = 0;
+    private int retryCounter = 0;
+
+    public RadioThermostatHandler(Thing thing, RadioThermostatStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing RadioThermostat handler.");
+
+        RadioThermostatConfiguration config = getConfigAs(RadioThermostatConfiguration.class);
+        logger.debug("config hostName = {}", config.hostName); 
+        logger.debug("config refresh = {}", config.refresh);
+        logger.debug("config logRefresh = {}", config.logRefresh);
+        logger.debug("config disableLogs = {}", config.disableLogs);
+        logger.debug("config disableHumidity = {}", config.disableHumidity);
+
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.hostName) == null) {
+            errorMsg = "Parameter 'hostName' is mandatory and must be configured";
+        }
+        if (config.refresh != null && config.refresh < 1) {
+            errorMsg = "Parameter 'refresh' must be at least 1 minute";
+        }
+        if (config.logRefresh != null && config.logRefresh < 5) {
+            errorMsg = "Parameter 'logRefresh' must be at least 5 minutes";
+        }
+        if (config.disableLogs != null && (config.disableLogs != 0 && config.disableLogs != 1)) {
+            errorMsg = "Parameter 'disableLogs' must be either 0 or 1";
+        }
+        if (config.disableHumidity != null && (config.disableHumidity != 0 && config.disableHumidity != 1)) {
+            errorMsg = "Parameter 'disableHumidity' must be either 0 or 1";
+        }
+
+        //test the thermostat connection by getting the thermostat name and model
+        if (errorMsg == null) {
+            try {
+                Object nameResult = getRadioThermostatData(NAME_RESOURCE);
+                if (nameResult instanceof RadioThermostatJsonName) {
+                    rthermData.setName(((RadioThermostatJsonName) nameResult).getName());
+                } else {
+                    errorMsg = "Unable to get thermostat name";
+                }
+                
+                Thread.sleep(2000);
+                
+                Object modelResult = getRadioThermostatData(MODEL_RESOURCE);
+                if (modelResult instanceof RadioThermostatJsonModel) {
+                    rthermData.setModel(((RadioThermostatJsonModel) modelResult).getModel());
+                } else {
+                    errorMsg = "Unable to get thermostat model";
+                }
+            } catch (Exception e) {
+                logger.error("Exception occurred attempting to connect with thermostat: {}", e.getMessage(), e);
```

Comment:
Bindings should only log to error, when they hit a bug. The correct log level would be warn. You also don't need to log the stacktrace when the connection fails, so remove the last parameter.

Reply:
Fixed and noted for future reference!